mpmprof

```diff
@@ -0,0 +1,310 @@
+#!/usr/bin/env python3
+"""
+Multiprocessing version of memory profiling of Python programs.
+"""
+
+import os
+import re
+import time
+import glob
+import argparse
+import subprocess
+import memory_profiler as mp
+
+from collections import defaultdict
+
+try:
+    import numpy as np
+    import matplotlib.pyplot as plt
+except ImportError:
+    plt = None
+    np  = None
+
+
+# Command Descriptions and Constants
+DESCRIPTION = "Multiprocessing memory profiling over time."
+EPILOG      = "If there are any bugs or concerns, submit an issue on Github"
+VERSION     = "mpmprof v{}".format(mp.__version__)
+FILETIME    = "%Y%m%d%H%M%S"
+BLANKS      = set(' \t')
+
+
+def run_action(args):
+    """
+    Run the given program and profile its memory usage.
+    """
+
+    # Determine where to write the output to
+    if args.output is None:
+        args.output = "mprofile_{}.dat".format(
+            time.strftime(FILETIME, time.localtime())
+        )
+
+    # Determine if the command is a Python command
+    if args.command[0].endswith('.py') and not args.nopython:
+        args.python = True
+
+    # Run the executable with the extra features
+    if args.python:
+        print("running as a Python program ...")
+        if not args.command[0].startswith('python'):
+            args.command.insert(0, 'python')
+
+    # Inform the user we're sampling
+    print("mpmprof: Sampling memory every {} seconds".format(args.interval))
+
+    # Put the command back together from the argument parsing
+    command = " ".join([
+        c if BLANKS.isdisjoint(c) else "'{}'".format(c) for c in args.command
+    ])
+
+    # Open a subprocess to the given command
+    proc = subprocess.Popen(args.command)
+
+    # This is where a call to mp.memory_usage should go.
+    # Instead we're adding the custom code for sampling spawned memory
+    with open(args.output, "a") as f:
+
+        # Write the command to the data file
+        f.write("CMDLINE {}\n".format(command))
+
+        # Continue sampling until the subprocess is over, counting lines
+        lines = 0
+        while True:
+            # Determine if the subprocess is still running
+            if proc.poll() is not None: break
+
+            # Collect memory usage of master program and write to profile
+            mem = mp._get_memory(proc.pid)
+            f.write("MEM {0:.6f} {1:.4f}\n".format(mem, time.time()))
+            lines += 1
+
+            # Collect memory usage of spawned children and write to profile
+            for idx, mem in enumerate(mp._get_child_memory(proc.pid)):
```

Comment:
So here is the line where I independently collect the memory of each child and add them to the .dat file with the "CHLD#" key. If `include_children=True` then "main" will be the total memory consumed, and if not, it will be only the main process, but all processes will be plotted with `independent_children=True` or whatever flag we come up with.